### DISCUSSION TITLE
API3 Questions

### DISCUSSION THREAD

**Discussion by justinbarry** [AugurProject/turbo#5] [Specs]
API3

Paying for it with the bond. 
TVL insurance payouts. How do you claim? What does it actually cover and do we care?

 To-do:

1. Confirm L2 providers they support.
2. Meet with their dev
3. The graph?

Are we supposed to pay upfront?

Things needed for market creation:
1. Can we have access to that data to help a creator crate a market? i.e. Can we feed templates with api3 data.
2. What kind of stats can we get our hands on? 
3. Can we put data on chain beforehand to create market?

Resolution:
1. Can we delay paying fees until the time of report?
2. How do they know what to report where?
3. Make it as automated as possible.
4. We do need open reporting in case api3 fails to report in a certain window.
5. Pull vs push - can they report directory to our contracts?
6. How does insurance work if there is a fuck up with data quality?
7. How do we ensure there is enough liquidity to exchange collateral for their token?
8. How does their payment work? Do we need to send collateral per complete set mint or destruction?

Market factory per data source….

Options?
1. Need to support WETH

**Comment by pgebheim** [comment]
## Paying for API3 Fees

- Based on current information -- it appears that Augur will need a pool of API3 tokens which are locked up within API3 in sufficienty quantity to cover the costs of getting the data from the APIs.
- Market creators already provide a REP bond. REP can be swapped for API3 in order to get tokens to deposit into this fund.
- Market fees (e.g. reporting fee) can be split into a pool which is also used to buy API3 tokens to deposit into the fund.
- The goal here would be to always have enough tokens locked up to cover the current costs of API usage.
